contracts/sdk/indexer/backfiller.ts

```diff
@@ -0,0 +1,233 @@
+    if (
+      tx.transaction.message
+        .programIds()
+        .every((pk) => !pk.equals(parserState.Bubblegum.programId))
+    ) {
+      continue;
+    }
+    if (tx.transaction.message.accountKeys.every((pk) => !pk.equals(treeKey))) {
+      continue;
+    }
+    if (tx.meta.err) {
+      continue;
+    }
+    await handleLogs(
+      nftDb,
+      {
+        err: null,
+        logs: tx.meta.logMessages,
+        signature: tx.transaction.signatures[0],
+      },
+      { slot: slot },
+      parserState,
+      startSeq,
+      endSeq
+    );
+  }
+}
+
+async function plugGaps(
+  connection: Connection,
+  nftDb: NFTDatabaseConnection,
+  parserState: ParserState,
+  treeId: string,
+  startSlot: number,
+  endSlot: number,
+  startSeq: number,
+  endSeq: number
+) {
+  const treeKey = new PublicKey(treeId);
+  for (let slot = startSlot; slot <= endSlot; ++slot) {
+    await plugGapsFromSlot(
+      connection,
+      nftDb,
+      parserState,
+      treeKey,
+      slot,
+      startSeq,
+      endSeq
+    );
+  }
+}
+
+async function fetchAndPlugGaps(
+  connection: Connection,
+  nftDb: NFTDatabaseConnection,
+  minSeq: number,
+  treeId: string,
+  parserState: ParserState
+) {
+  let [missingData, maxDbSeq, maxDbSlot] = await nftDb.getMissingData(
+    minSeq,
+    treeId
+  );
+  let currSlot = await connection.getSlot("confirmed");
+
+  let merkleAccount = await connection.getAccountInfo(
+    new PublicKey(treeId),
+    "confirmed"
+  );
+  let merkleRoll = decodeMerkleRoll(merkleAccount.data);
+  let merkleSeq = merkleRoll.roll.sequenceNumber.toNumber() - 1;
+
+  if (merkleSeq - maxDbSeq > 1 && maxDbSeq < currSlot) {
+    console.log("Running forward filler");
+    missingData.push({
+      prevSeq: maxDbSeq,
+      currSeq: merkleSeq,
+      prevSlot: maxDbSlot,
+      currSlot: currSlot,
+    });
+  }
+
+  let backfillJobs = [];
+  for (const { prevSeq, currSeq, prevSlot, currSlot } of missingData) {
+    console.log(prevSeq, currSeq, prevSlot, currSlot);
+    backfillJobs.push(
+      plugGaps(
+        connection,
+        nftDb,
+        parserState,
+        treeId,
+        prevSlot,
+        currSlot,
+        prevSeq,
+        currSeq
+      )
+    );
+  }
+  if (backfillJobs.length > 0) {
+    await Promise.all(backfillJobs);
+  }
+  return maxDbSeq;
+}
+
+async function main() {
+  const endpoint = localhostUrl;
+  const connection = new Connection(endpoint, "confirmed");
+  const payer = Keypair.generate();
+  const provider = new anchor.Provider(connection, new NodeWallet(payer), {
+    commitment: "confirmed",
+  });
+  let nftDb = await bootstrap(false);
+  Gummyroll = loadProgram(
+    provider,
+    GUMMYROLL_PROGRAM_ID,
+    "target/idl/gummyroll.json"
+  ) as anchor.Program<Gummyroll>;
+  Bubblegum = loadProgram(
+    provider,
+    BUBBLEGUM_PROGRAM_ID,
+    "target/idl/bubblegum.json"
+  ) as anchor.Program<Bubblegum>;
+  while (true) {
+    for (const [treeId, depth] of await nftDb.getTrees()) {
```

Comment:
We probably only want to listen to trees that we list in a Config, then hot-reload from that config.

Config can be editable from a CLI